ref(sentry apps): Reorganize UI components info (#4994)
getsentry/sentry-docs: src/docs/product/integrations/integration-platform/ui-components/formfield.mdx

---
title: FormField Components
sidebar_order: 4
description: "Learn about FormField components."
---

> The issue linking and alert rule action UI components can utilize `FormField` components to display attributes in their respective modals. A `FormField` can be one of three types (which behave like their HTML equivalents) — `Text`, `TextArea,` and `Select`.

Reviewer: I wouldn't mention specific UI components here in case we add more that accept FormFields. Maybe only as examples:

> Some UI Components (e.g. Issue Linking, Alert Rule Action) can render custom forms with inputs specified according to the schemes below. They behave similar to their HTML equivalents.

Reviewer: I also wouldn't list out Text, TextArea and Select, since it'd be on someone to remember or notice they must update this sentence if we add another field type later on.

> The issue linking and alert rule action UI components can utilize `FormField` components to display attributes in their respective modals. Some UI Components (e.g. [Issue Link](/product/integrations/integration-platform/ui-components/issue-link), [Alert Rule Actions](/product/integrations/integration-platform/ui-components/alert-rule-action)) can render custom forms with inputs specified according to the schemes below. They behave similarly to their HTML equivalents.

Reviewer 2: Streamlined this because it was repetitive

```suggestion
The [issue linking](/product/integrations/integration-platform/ui-components/issue-link) and [alert rule action](/product/integrations/integration-platform/ui-components/alert-rule-action) UI components can use `FormField` components to display attributes in their respective modals. These components can render custom forms with inputs specified according to the schemes below. They behave similarly to their HTML equivalents.
```

## Text

### Schema

```javascript
{
    "type": "text",
    "label": <String>,
    "name": <String>,
    (optional) default: <String>,
}
```

### Attributes

- `label` - (Required) Label text to be rendered for the form field
- `name` - (Required) Value to use in `name` attribute of the field

> - `default` - default to pre-populate with. Options include `issue.title` and `issue.description`

Reviewer: Idk what it should look like but this needs a note that it only applies to Issue linking

  - `issue.title` - title of the Sentry issue
  - `issue.description` - description of the Sentry issue

## Textarea

### Schema

```javascript
{
    "type": "textarea",
    "label": <String>,
    "name": <String>,
    (optional) default: <String>,
}
```

### Attributes

- `label` - (Required) Label text to be rendered for the form field
- `name` - (Required) Value to use in `name` attribute of the field

> - `default` - default to pre-populate with options include `issue.title` and `issue.description`

Reviewer: See above

  - `issue.title` - title of the Sentry issue
  - `issue.description` - description of the Sentry issue

## Select

### Schema

```json {filename:schema.json}
{
  "type": "select",
  "label": <String>,
  "name": <String>,
  "uri": <URI>,
  "async": <Bool>,
  "options": <Array<String, String>>,
  "depends_on": <Array<String>> (optional)

>   "skip_load_on_open": <Array<String>> (optional)

Reviewer: Can we remove `(optional)` from the codeblocks? It's not consistent with the text below, and it looks more consistent without em.

}
```

### Attributes

- `label` - (Required) Label text to be rendered for the form field
- `name` - (Required) Value to use in `name` attribute of the field
- `uri` - (Required if developer doesn't provide `options`) URI to retrieve values from.
- `async` - (Optional) Used only if `uri` is present. If true (default), will query the URI as the user types, for autocomplete suggestions (see response format below). If false, will query the URI once initially to retrieve all possible options. This request _must_ succeed, and the response _must_ return data in the format Sentry expects, otherwise the entire component won't render.
- `options` - (Required if developer doesn't provide `uri`) Static list of options in the format of [value, label]
- `depends_on` - (Optional) If a field value depends on the value of another field, a request will be made to load those options when the dependent field is set.
- `skip_load_on_open` (Optional) If true, this field will not load options when the page loads. Instead, it will wait for the user to start typing before fetching options.

### URI Request Format

> The request sent to the `uri` to load options is a `GET` request. The parameters are encoded into the query parameters as such:

Reviewer 2: ```suggestion
The request sent to the `uri` to load options is a `GET` request. The parameters are encoded into the query parameters like so:
```

```javascript
{
    "installationId": <String>,
    "projectSlug": <String>,
    "query": <String>,
    "dependentData": <String> (optional),
}
```

- `installationId` - The ID of the installation associated with the request
- `projectSlug` - The slug of the project in Sentry
- `query` - The search term the user is typing
- `dependentData` - A JSON encoded string of the dependent data if that field has `depends_on` set

### URI Response Format

> Response from `uri`, when specified, _must_ be in the following format:

Reviewer: I'd prefer
> must be JSON and match the following format:

>   "dependentData": <String> (optional),

Reviewer: See comment above

> ```javascript

Reviewer: json

[
  {
    "label": <String>,
    "value": <String>,

>     "default": <Bool>, (optional)

Reviewer: See comment above
